recipes-devtools/opkg-utils/files/0001-add-metadata-parsing-functionality-to-Package-class.patch

```diff
@@ -0,0 +1,66 @@
+From 234a35bfdb3b61184a61dff1c480b3d2d8debf5d Mon Sep 17 00:00:00 2001
+From: pratheekshasn <pratheeksha.s.n@ni.com>
+Date: Mon, 9 Dec 2024 13:08:04 +0530
+Subject: [PATCH] add metadata parsing functionality to Package class
+
+Upstream-Status: Inappropriate [nilrt-specific]
+
+Signed-off-by: pratheekshasn <pratheeksha.s.n@ni.com>
+---
+ opkg.py | 22 ++++++++++++++++++++++
+ 1 file changed, 22 insertions(+)
+
+diff --git a/opkg.py b/opkg.py
+index e5a65dd..c181421 100644
+--- a/opkg.py
++++ b/opkg.py
+@@ -182,6 +182,8 @@ class Package(object):
+         self.tags = None
+         self.fn = fn
+         self.license = None
++        self.user_visible = None
++        self.display_name = None
+ 
+         self.user_defined_fields = collections.OrderedDict()
+         if fn:
+@@ -291,6 +293,10 @@ class Package(object):
+                     self.sha256 = value
+                 elif name_lowercase in self.__dict__:
+                     self.__dict__[name_lowercase] = value
++                elif name_lowercase == 'uservisible':
++                    self.user_visible = value
++                elif name_lowercase == 'displayname':
++                    self.display_name = value
```

Comment:
Rather than carry an out-of-tree patch for this change, I imagine you could implement it as a list of custom fields that you can define at runtime. And then we can upstream this patch. (I am the upstream maintainer.)

Reply:
I have fixed this now.